src/mechanics/constitutive/LocalIsotropicDamage.h

```diff
@@ -0,0 +1,137 @@
+        auto sigma = mElasticLaw.Stress(strain, deltaT, cellId, ipId);
+
+        auto kappa = mEvolution.Kappa(strain, deltaT, cellId, ipId);
+        auto omega = mDamageLaw.Damage(kappa);
+        auto dOmegadKappa = mDamageLaw.Derivative(kappa);
+
+        return (1 - omega) * C - sigma * dOmegadKappa * mEvolution.DkappaDstrain(strain, deltaT, cellId, ipId);
+    }
+
+    void Update(EngineeringStrain<TDim> strain, double deltaT, int cellId, int ipId)
+    {
+        mEvolution.Update(strain, deltaT, cellId, ipId);
+    }
+
+public:
+    // Intentionally made public. I assume that we know what we do. And this avoids lots of code, either getters
+    // (possibly nonconst) or forwarding functions like `double Kappa(...), double Damage(...)`
+    TDamageLaw mDamageLaw;
+    TEvolution mEvolution;
+    TElasticLaw mElasticLaw;
+};
+
+template <int TDim, typename TDamageLaw, typename TStrainNorm, typename TElasticLaw>
+static auto CreateLocalIsotropicDamage(TDamageLaw damageLaw, TStrainNorm strainNorm, TElasticLaw elasticLaw)
+{
+    return LocalIsotropicDamage<TDim, TDamageLaw, TStrainNorm, TElasticLaw>(damageLaw, strainNorm, elasticLaw);
+}
+
+/** Explicit evolution equation for the NuTo::LocalIsotropicDamageLaw that implements
+ * \f[
+ *  \kappa(\boldsymbol \varepsilon) = \max \left(\kappa, \varepsilon_\text{eq}(\boldsymbol \varepsilon) \right)
+ * \f]
+ *
+ * @tparam TDim dimension
+ *
+ * @tparam TStrainNorm strain norm \f$ \varepsilon_\text{eq}(\boldsymbol \varepsilon) \f$. This requires the methods
+ * `Kappa(strain)` and `DkappaDstrain(strain)` to be implemented, e.g. NuTo::Constitutive::ModifiedMises.
+ */
+template <int TDim, typename TStrainNorm>
+class EvolutionImplicit
+{
+
+public:
+    //! Constructor. As this evolution equation requires history data, they are also allocated.
+    //! @param strainNorm strain norm, see class documentation
+    //! @param numCells number of cells for the history data allocation
+    //! @param numIpsPerCell nummer of integraiton points per cell for the history data allocation
+    EvolutionImplicit(TStrainNorm strainNorm, size_t numCells, size_t numIpsPerCell)
+        : mStrainNorm(strainNorm)
+        , mNumIpsPerCell(numIpsPerCell)
+        , mKappas(std::vector<double>(numCells * numIpsPerCell, 0.))
+    {
+    }
+
+    double Kappa(EngineeringStrain<TDim> strain, double, int cellId, int ipId) const
+    {
+        return std::max(mStrainNorm.Value(strain), mKappas[Ip(cellId, ipId)]);
+    }
+
+    double DkappaDstrain(EngineeringStrain<TDim> strain, double, int cellId, int ipId) const
+    {
+        if (mStrainNorm.Value(strain) > mKappas[Ip(cellId, ipId)])
+            return 1.;
+        return 0;
+    }
+
+    void Update(EngineeringStrain<TDim> strain, double deltaT, int cellId, int ipId)
+    {
+        mKappas[Ip(cellId, ipId)] = Kappa(strain, deltaT, cellId, ipId);
+    }
+
+    size_t Ip(int cellId, int ipId) const
+    {
+        return cellId * mNumIpsPerCell + ipId;
```

Comment:
Have we agreed on making separate integrands for cells of different dimensionality of the integrand (e.g. triangles and quads with a different number of integration points per cell)

Reply:
`THistoryDataContainer` with `THistoryDataContainer::operator(cellId, ipId)` could be another template parameter for that.